apps/users/src/app/app.controller.ts

```diff
@@ -651,4 +651,87 @@ export class UserController {
 
     await this.userService.reportUser(user, reportedUser, message);
   }
+
+  @UsePipes(new ValidationPipe({ skipMissingProperties: true }))
+  @ApiBody({
+    type: UpdateMobileDto
+  })
+  @ApiOkResponse({
+    type: UserResponseDto
+  })
+  @CastleClearCacheAuth(CacheKeyName.Users)
+  @Put('me/mobile')
+  async updateMobile(
+    @Req() req: CredentialRequest,
+    @Body() body: UpdateMobileDto
+  ) {
+    const objective: OtpObjective = <OtpObjective>body.objective;
+    if (
+      !objective ||
+      !Object.values(OtpObjective).includes(objective) ||
+      objective !== OtpObjective.VerifyMobile
+    ) {
+      logger.error(`Invalid objective.`);
+      throw new CastcleException(
+        CastcleStatus.PAYLOAD_TYPE_MISMATCH,
+        req.$language
+      );
+    }
+
```

Comment:
```suggestion
```

These lines are unused since `body.objective` has been validated in `UpdateMobileDto`.